### ISSUE TITLE
push all commits unexpected error in 1.0

### ISSUE THREAD

**Issue by dmpetrov** [iterative/dvc#3812]
```bash
$ dvc -V
1.0.0a1+b8aa8a
```

```bash
$ dvc push --all-commits
ERROR: unexpected error
```

```bash
$ dvc push --all-commits -v
2020-05-18 02:52:04,782 DEBUG: PRAGMA user_version;
2020-05-18 02:52:04,783 DEBUG: fetched: [(3,)]
2020-05-18 02:52:04,783 DEBUG: CREATE TABLE IF NOT EXISTS state (inode INTEGER PRIMARY KEY, mtime TEXT NOT NULL, size TEXT NOT NULL, md5 TEXT NOT NULL, timestamp TEXT NOT NULL)
2020-05-18 02:52:04,783 DEBUG: CREATE TABLE IF NOT EXISTS state_info (count INTEGER)
2020-05-18 02:52:04,783 DEBUG: CREATE TABLE IF NOT EXISTS link_state (path TEXT PRIMARY KEY, inode INTEGER NOT NULL, mtime TEXT NOT NULL)
2020-05-18 02:52:04,783 DEBUG: INSERT OR IGNORE INTO state_info (count) SELECT 0 WHERE NOT EXISTS (SELECT * FROM state_info)
2020-05-18 02:52:04,784 DEBUG: PRAGMA user_version = 3;
2020-05-18 02:52:04,788 DEBUG: Assuming '/Users/dmitry/src/viewer-metrics-gen/repo-johnnie/.dvc/cache/52/a6e6a3af83f1bd127a249cec82d9fb.dir' is unchanged since it is read-only
2020-05-18 02:52:04,788 DEBUG: Assuming '/Users/dmitry/src/viewer-metrics-gen/repo-johnnie/.dvc/cache/52/a6e6a3af83f1bd127a249cec82d9fb.dir' is unchanged since it is read-only
2020-05-18 02:52:04,833 DEBUG: SELECT count from state_info WHERE rowid=?
2020-05-18 02:52:04,833 DEBUG: fetched: [(221,)]
2020-05-18 02:52:04,834 DEBUG: UPDATE state_info SET count = ? WHERE rowid = ?
2020-05-18 02:52:04,836 ERROR: unexpected error
------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/dmitry/src/dvc/dvc/main.py", line 48, in main
    ret = cmd.run()
  File "/Users/dmitry/src/dvc/dvc/command/data_sync.py", line 58, in run
    run_cache=self.args.run_cache,
  File "/Users/dmitry/src/dvc/dvc/repo/__init__.py", line 28, in wrapper
    ret = f(repo, *args, **kwargs)
  File "/Users/dmitry/src/dvc/dvc/repo/push.py", line 32, in push
    used_run_cache=used_run_cache,
  File "/Users/dmitry/src/dvc/dvc/repo/__init__.py", line 290, in used_cache
    filter_info=filter_info,
  File "/Users/dmitry/src/dvc/dvc/stage/__init__.py", line 513, in get_used_cache
    cache.update(out.get_used_cache(*args, **kwargs))
  File "/Users/dmitry/src/dvc/dvc/output/base.py", line 452, in get_used_cache
    self.checksum, self._collect_used_dir_cache(**kwargs),
  File "/Users/dmitry/src/dvc/dvc/output/base.py", line 366, in _collect_used_dir_cache
    if self.cache.changed_cache_file(self.checksum):
  File "/Users/dmitry/src/dvc/dvc/remote/base.py", line 804, in changed_cache_file
    if self.is_protected(cache_info):
  File "/Users/dmitry/src/dvc/dvc/remote/local.py", line 711, in is_protected
    if not self.exists(path_info):
  File "/Users/dmitry/src/dvc/dvc/remote/local.py", line 95, in exists
    assert is_working_tree(self.repo.tree)
AssertionError
------------------------------------------------------------
```

**Comment by efiop** [comment]
Would be automatically solved (or close to it) by https://github.com/iterative/dvc/issues/3811

**Comment by pmrowla** [comment]
This is fixed after the changes to #3811, but we should have test cases for push with `all-commits`/`all-branches`/`all-tags` enabled
